### PR TITLE
Added foreach function

### PR DESCRIPTION
**Pull request by Vijaykv5**
#3118 

### Description
Added the foreach() function as part of the tables functions terms.
### Type of Change

created foreach.md file

### Checklist

- [ ] All writings are my own.
- [ ] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [ ] I have performed a self-review of my own writing and code.
- [ ] I have checked my entry and corrected any misspellings.
- [ ] I have made corresponding changes to the documentation if needed.
- [ ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

@SSwiniarski  Please Check

### CODE REVIEW

**Comment by SSwiniarski** [comment]
@Vijaykv5 Two things before this gets reviewed. You need to add the missing metadata section at the start of your file. You can get a summary on how to put that together in the style guide [here](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#metadata). You also need to sign the [CLA](https://cla-assistant.io/Codecademy/docs?pullRequest=3318). Tag me when this has been addressed and I'll do a first maintainer review.

**Comment by Vijaykv5** [comment]
@SSwiniarski I've added the metadata and also Signed the CLA Agreement

File: content/lua/concepts/tables/terms/foreach/foreach.md
Comment (SSwiniarski) [inline content/lua/concepts/tables/terms/foreach/foreach.md:11]:
```suggestion
---
Title: 'foreach()'
Description: 'Iterates through the elements of a provided table or collection, using a specified callback function to perform operations on each element'
Subjects:
  - 'Computer Science'
Tags:
  - 'Tables'
  - 'Arrays'
CatalogContent:
  - 'learn-lua'
  - 'paths/computer-science'
---
```
The metadata had a number of issues that I corrected. Most important is this entry has nothing to do with JavaScript.

File: content/lua/concepts/tables/terms/foreach/foreach.md
Comment (SSwiniarski) [inline content/lua/concepts/tables/terms/foreach/foreach.md:14]:
```suggestion

```
Entries do not have a title header. The title in the metadata takes care of that.

File: content/lua/concepts/tables/terms/foreach/foreach.md
Comment (SSwiniarski) [inline content/lua/concepts/tables/terms/foreach/foreach.md:35]:
```suggestion
table.foreach(fruits, printFruit)
```

File: content/lua/concepts/tables/terms/foreach/foreach.md
Comment (SSwiniarski) [inline content/lua/concepts/tables/terms/foreach/foreach.md:23]:
```suggestion
In this example is a table containing the names of fruits, and it will use `.foreach()` to print each fruit name along with its index in the table.
```
We avoid first and second-person pronouns in docs.

File: content/lua/concepts/tables/terms/foreach/foreach.md
Comment (SSwiniarski) [inline content/lua/concepts/tables/terms/foreach/foreach.md:44]:
````suggestion

This example results in the following output:

```shell
Fruit at index 1: apple
Fruit at index 2: banana
Fruit at index 3: cherry
Fruit at index 4: date
Fruit at index 5: elderberry
```
````

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Vijaykv5 I did a review and left you a few comments.

**Comment by SSwiniarski** [comment]
@yangc95, @KTom101, @caupolicandiaz Is this another one that should be under Luau rather than Lua? I tested the example in a online Lua interpreter and it failed, but in a Luau one it worked.

**Comment by Vijaykv5** [comment]
changed the files as per the suggesions @SSwiniarski

File: content/lua/concepts/tables/terms/foreach/foreach.md
Comment (SSwiniarski) [inline content/lua/concepts/tables/terms/foreach/foreach.md:8]:
```suggestion
  - 'Tables'
  - 'Arrays'
```

File: content/lua/concepts/tables/terms/foreach/foreach.md
Comment (SSwiniarski) [inline content/lua/concepts/tables/terms/foreach/foreach.md:47]:
````suggestion
```shell
Fruit at index 1: apple
Fruit at index 2: banana
Fruit at index 3: cherry
Fruit at index 4: date
Fruit at index 5: elderberry
```
````

File: content/lua/concepts/tables/terms/foreach/foreach.md
Comment (SSwiniarski) [inline content/lua/concepts/tables/terms/foreach/foreach.md:21]:
````suggestion
### Syntax 
 
```pseudo
foreach(table, callback)
```

### Example
````
Also, add a sentence after the syntax explaining what `table` and `callback` are.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Vijaykv5 some more comments.

**Comment by caupolicandiaz** [comment]
> @yangc95, @KTom101, @caupolicandiaz Is this another one that should be under Luau rather than Lua? I tested the example in a online Lua interpreter and it failed, but in a Luau one it worked.

I believe this is a Luau only method

**Comment by SSwiniarski** [comment]
@Vijaykv5 can we change the location of the file from `content/lua/concepts/tables/terms/foreach/foreach.md` to `content/luau/concepts/tables/terms/foreach/foreach.md`

**Comment by Vijaykv5** [comment]
@SSwiniarski Made the changes as per your comments

File: content/luau/concepts/tables/terms/foreach/foreach.md
Comment (SSwiniarski) [inline content/luau/concepts/tables/terms/foreach/foreach.md:24]:
```suggestion
 
 - `table` refers to a data structure, such as an array or list, that contains elements to iterate over.
- `callback` is a function provided as an argument to the loop, which gets executed for each element in `table`, typically to perform a specific action or operation on each item.

### Example
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Vijaykv5 one more comment.

**Comment by Vijaykv5** [comment]
Check now please @SSwiniarski

**Comment by Vijaykv5** [comment]
@SSwiniarski May you can check it

File: content/luau/concepts/tables/terms/foreach/foreach.md
Comment (SSwiniarski) [reply content/luau/concepts/tables/terms/foreach/foreach.md:24]: @Vijaykv5 you still need to address this.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Vijaykv5 you still need to address the last comment.

**Comment by Vijaykv5** [comment]
@SSwiniarski Please check it now :) 
Also may you can assign me under hacktoberfest

**Review by SSwiniarski — APPROVED** [review]
@Vijaykv5 I cleaned up some formatting and this is ready for a second review.

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@Vijaykv5 I've added a few additional comments. Please update, and tag me when you've made the changes. Thanks

**Review by caupolicandiaz — APPROVED** [review]
@Vijaykv5 @SSwiniarski I made a couple of direct edits to wrap this up. Good to merge

**Pull request merged**
